Fix integration tests for Ansible 2.19.

burnison opened the pull request:
##### SUMMARY
This commit makes the necessary changes to run integration tests with Ansible 2.16 through Ansible 2.19. Because Ansible 2.19 does not work with any Ubuntu release that supports Percona Server 5.7, this commit upgrades the `percona-server-client` to version 8.0, which is currently the oldest supported release of Percona Server.

This commit was large enough that it didn't fix the Ansible 2.19 warnings, just the errors, but does begin to address #174.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
All integration tests.

burnison committed:
Fix integration tests for Ansible 2.19.

This commit makes the necessary changes to run integration tests with
Ansible 2.16 through Ansible 2.19. Because Ansible 2.19 does not work
with any Ubuntu release that supports Percona Server 5.7, this commit
upgrades the `percona-server-client` to version 8.0, which is currently
the oldest supported release of Percona Server.

This commit was large enough that it didn't fix the Ansible 2.19
warnings, just the errors.

burnison commented:
Locally, I had to run `ansible-test` with `--docker-privileged` to allow ProxySQL service to run on Ansible 2.18 and 2.19. I'm uncertain if there's a similar mechanism in the Github Action runner, if that's even the problem.

burnison committed:
Coerce types for python3.12+ansible2.17.

Some tests in `proxysql_global_variables` compared strings to integers,
which doesn't seem to work in all permutations of python+ansible. By
coercing the comparisons to strings, the issue is prevented.

Andersson007 commented:
@burnison hello, thanks for the PR!
1. You can remove 2.16 from matrix and not test against it because it's EOL
2. If there are any unnecessary changes like cosmetic playbook syntax changes in this PR (I'm not sure), I would suggest moving them to a separate PR
3. I think we can see what CI actually runs under the `.github/...` directory

burnison commented:
I do not believe there are any unnecessary cosmetic changes.

I believe the underlying failure in CI is the same I had locally: ProxySQL requires privileged capabilities to run that `ansible-test` doesn't, or no longer, provides as of 2.18 (I can lookup the exact capability that was logged in journald later). Locally, the service only starts when `ansible-test` was run with `--docker-privileged` (as I couldn't see a way to pass granular capabilities). With that, all tests in the matrix should pass.

Andersson007 commented:
@burnison https://github.com/ansible-collections/community.proxysql/blob/97c8e096ba03ef4c16ef48a77802dbe0017e0d28/.github/workflows/ansible-test-plugins.yml#L70
You can see there's the `with` directive. You could research around that custom action, maybe there's a directive that we can use to pass `--docker-privileged`?

Andersson007 commented:
I remember there should be some docs in the action's repo somewhere, maybe in the form of comments
IMPORTANT: I'm not a maintainer of this repo. Trying to help figure out how to fix it

burnison commented:
@Andersson007, I tried to work-around this in 50ef720306766717f649461f964266e45a56d0a8, but I removed it from the PR because `python-3.12` isn't available in `ubuntu2204`.

The gist seems to be that `ubuntu2404` prevents ProxySQL from starting due to the missing `CAP_SYS_ADMIN` capability (which is why `--docker-privileged` works). It _seems_ this is blocked on ansible-community/ansible-test-gh-action#103, though there may be other work-arounds I can't find. I'll spend a bit more time looking, but it may be blocked for now.

burnison committed:
Create systemd overrides for integration tests.

In Ubuntu 24.04, the combination of `PrivateDevices` and
`RestrictAddressFamilies` was leading to a need for the `CAP_SYS_ADMIN`
capability within the docker container. These stanzas within systemd are
intended to sandbox processes, reducing the attack surface of units run
within it.

While I think both of these systemd stanzas may make sense for
production deployments, I am not sure they make sense on ephemeral test
containers --- especially if it requires additional system privileges by
running the container in privileged mode.

This commit adds a systemd override to remove both systemd
configurations for the unit.

burnison commented:
@Andersson007, I found a work-around that doesn't require privilege escalation. All tests are passing in CI now.

Andersson007 commented:
@burnison thanks a lot for the contribution, it has saved the collection!
Any other contributions from you here and everywhere across the Ansible ecosystem are very welcome!
Please contact me if you have any questions
